js_modules/dagit/packages/ui/src/components/ProductTour.tsx

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import {createPortal} from 'react-dom';
+import styled, {CSSProperties} from 'styled-components/macro';
+
+import {Box} from './Box';
+import {Button} from './Button';
+import {Colors} from './Colors';
+import {Subheading} from './Text';
+
+export enum Position {
+  TOP_LEFT,
+  TOP_CENTER,
+  TOP_RIGHT,
+  BOTTOM_LEFT,
+  BOTTOM_CENTER,
+  BOTTOM_RIGHT,
+}
+
+type ObjectType =
+  | {img: string; video?: undefined; object?: undefined}
+  | {video: string; img?: undefined; object?: undefined}
+  | {object: React.ReactNode; video?: undefined; img?: undefined}
+  | {img?: undefined; video?: undefined; object?: undefined};
+
+type Props = {
+  title: React.ReactNode;
+  description?: React.ReactNode;
+  actions?: {
+    next?: () => void;
```

Comment:
Any need for `prev`?

Reply:
I don't think so right now, it's flexible to let you add that if you need it though.